xwiki-platform-core/xwiki-platform-notifications/xwiki-platform-notifications-api/src/main/java/org/xwiki/notifications/CompositeEvent.java

```diff
@@ -54,6 +54,17 @@ public CompositeEvent(Event event)
     }
 
     /**
+     * Copy constructor of {@link CompositeEvent}.
+     *
+     * @param compositeEvent the composite event to copy
+     */
+    public CompositeEvent(CompositeEvent compositeEvent)
+    {
+        this.events = compositeEvent.events;
```

Comment:
It means you will the same list. So if you change `this.events`, it will impact `compositeEvent.events` too.